### ISSUE TITLE
highlight the changes left to sync

### ISSUE THREAD

**Issue by IndianBoy42** [MagicDuck/grug-far.nvim#65]
Hey! love the edit in search results panel then sync it back feature. I think it could be even better if there was some highlighting of the diff between whats in the panel and whats in the files.

**Comment by MagicDuck** [comment]
Unfortunately, the calculation of that diff and application of the highlight would have to happen on every text change event. I think this could be prohibitively expensive performance wise as you would basically be doing a poor mans treesitter on the results listing.

**Comment by IndianBoy42** [comment]
diffing is not that expensive, nowhere close to doing a treesitter parse. and most of the time the number of highlights to apply would be pretty small. I think its worth try, I may attempt it this weekend

**Comment by MagicDuck** [comment]
Hmm, maybe if you wrap it in a `vim.schedule` and do it in chunks so it does not block input?. Idk I am still pretty wary of doing this on a buffer where there are thousands of lines of results. I've had the experience of opening large files in treesitter before, haha. What would you show for deleted text?

**Comment by IndianBoy42** [comment]
Have you tried gitsigns diff mode or diffview? Diffing is super optimized so I'm pretty sure if we use `vim.diff` it calls a c routine that just goes through it for us. Treesitter is doing way more complicated work: state machines and branching and regex and everything. Diffing is a pretty straight shot linear comparison. 

I think either a little red highlight around the position the deleted text was, or the deleted text itself but highlighted in red. Probably keep it similar to gitsigns.

**Comment by MagicDuck** [comment]
I haven't tried those but am assuming those still operate on smaller files. 
Treesitter does not do regex as far as I know. That is the whole idea, it has C code for each file type that parses/tokenizes the files and generates an AST.

For our case, I guess we could do straight string comparison if the assumption is that the amount of places manually changed is small. 

`vim.diff` does not give me anything very useful though. For example:
```
= vim.diff('something one', 'something extra one', {result_type = 'indices'})

--> { { 1, 1, 1, 1 } } 
```
(same result for removing the text)

> I think either a little red highlight around the position the deleted text was, or the deleted text itself but highlighted in red. Probably keep it similar to gitsigns.
Yeah, I want to keep the text as close as possible to what rg generated, so don't want to have the deleted text in there.

So I think the most we could show would be a gitsigns style vertical bar or somesuch in the sign column. Essentially an indicator that you edited that text.

**Comment by MagicDuck** [comment]
Ok, so I luckily happened to watch a video by tjdevries that talked about the `vim.api.nvim_buf_attach()` function and that turned out to be what was needed to fix our perf issues since it tells you which lines changed on every change event. So I can incrementally update the change indicators.

<img width="928" alt="image" src="https://github.com/MagicDuck/grug-far.nvim/assets/95201/8157dc18-a3d1-49d0-947e-49b0011763f9">

The change is in: https://github.com/MagicDuck/grug-far.nvim/pull/73
Note that there's a new icon and highlight for the indicator in the sign column. I tried to make it look like gitsigns 😄 
Closing as it's addressed, but feel free to provide further feedback!
